aclk/aclk_query.c

```diff
@@ -14,6 +14,8 @@ pthread_mutex_t query_lock_wait = PTHREAD_MUTEX_INITIALIZER;
 #define QUERY_THREAD_LOCK pthread_mutex_lock(&query_lock_wait)
 #define QUERY_THREAD_UNLOCK pthread_mutex_unlock(&query_lock_wait)
 
+__thread struct aclk_query_thread *aclk_query_thread = NULL;
```

Comment:
although generally cleaner solution I worry about performance impact here as opposed to passing the pointer to function from above. Will this translate to `pthread_getspecific`?

Reply:
if I remember correctly when I implement this I do it this way exactly to avoid the `pthread_getspecific` as the thread is quite simple and passing this pointer from thread main was not much of an issue